### PR TITLE
Use identity<...>::type in the right place.

### PR DESCRIPTION
**Pull request by bangerth**
Follow-up to #8712. @marcfehling: FYI.

### CODE REVIEW

File: include/deal.II/hp/refinement.h [hunk 1]
```diff
@@ -207,8 +206,10 @@ namespace hp
       const Vector<Number> &               criteria,
       const Number                         p_refine_threshold,
       const Number                         p_coarsen_threshold,
-      const ComparisonFunction<Number> &compare_refine = std::greater<Number>(),
-      const ComparisonFunction<Number> &compare_coarsen = std::less<Number>());
+      const ComparisonFunction<typename identity<Number>::type>
+        &compare_refine = std::greater<Number>(),
+      const ComparisonFunction<typename identity<Number>::type>
+        &compare_coarsen = std::less<Number>());
```
Comment (marcfehling) [inline include/deal.II/hp/refinement.h]: Aha! This way around. This makes sense to me now.

**Comment by marcfehling** [comment]
```
/jenkins/workspace/dealii_PR-8987/tests/hp/error_prediction.cc:105: error: undefined reference to 'void dealii::hp::Refinement::p_adaptivity_from_reference<1, float, 1>(dealii::hp::DoFHandler<1, 1> const&, dealii::Vector<float> const&, dealii::Vector<float> const&, std::function<bool (dealii::identity<float>::type const&, dealii::identity<float>::type const&)> const&, std::function<bool (dealii::identity<float>::type const&, dealii::identity<float>::type const&)> const&)'
```

I guess you need to update the `.cc` file accordingly, too.

**Comment by bangerth** [comment]
Ah, yes. Let's see whether this works!

**Pull request merged**